editor/editor_node.cpp

```diff
@@ -3423,12 +3423,13 @@ int EditorNode::_next_unsaved_scene(bool p_valid_filename, int p_start) {
 
 void EditorNode::_exit_editor(int p_exit_code) {
 	exiting = true;
-	resource_preview->stop(); // Stop early to avoid crashes.
```

Comment:
This needs _really_ thorough testing. The resource previewer is one of the most crash, deadlock and regression prone code in the editor - as this comment points out. Running `Main::iteration()` explicitly is also almost always a recipe for disaster. Please quadruple check that this is working well in complex scenarios (with and without resource previews pre-existing in the user cache folder).

Reply:
- Regarding how early the previewer has to be stopped, that code and comment were added in #24206. Formerly, the previewer was stopped on destruction, much later. And that of course would summon many demons. My assessment is that it's safe to do the controlled stop anywhere in this function. However, I just intended the editor dim to happen earlier, so I'll play it supersafe and just keep this function intact.
- About `Main::iteration()`, the only calls needed from it are `RenderingServer::sync()` and `MessageQueue::flush()`. Those are the currently known heartbeats that keep rendering and resource loading reactive. I chose to run an iteration so this code doesn't get outdated whenever there's a new requirement in that regard. However, now we're aware, I'm changing this to is supersafe: the code will do both calls explicitly instead of invoking a full iteration.

I'm pushing both changes. Verified the issue is still fixed after removing `.godot/`.

Thanks for the warning, because we definitely will enjoy a regression-freer period. 😀